job_definitions/notify_buyers_to_award_closed_briefs.yml

```diff
@@ -0,0 +1,81 @@
+---
+{% for environment in environments %}
+{% with vars = [
+    {"offset": "28", "weeks": "4", "notify_template_id": "204d2bf5-5dd3-44c8-a68f-9111e040e831"},
+    {"offset": "56", "weeks": "8", "notify_template_id": "bc1f93a8-9529-48b6-9a82-af61bc6da332"}
+  ]
+%}
+{% for v in vars %}
+- job:
+    name: "notify-buyers-to-award-closed-briefs-{{ v.weeks }}-weeks-{{ environment }}"
+    display-name: "Notify buyers to award closed briefs ({{ v.weeks }} weeks) - {{ environment }}"
+    project-type: freestyle
+    description: "Send first email notification to buyer users to award closed briefs, after 4 weeks."
+    parameters:
+      - string:
+          name: NOTIFY_TEMPLATE_ID
+          default: "{{ v.notify_template_id }}"
+          description: "Notify template ID for {{ v.weeks }}-week email"
+      - string:
+          name: OFFSET_DAYS
+          default: "{{ v.offset }}"
+          description: "Notify buyers for briefs that closed X days ago"
+      - bool:
+          name: DRY_RUN
+          default: false
+          description: "List notifications that would be sent without sending the emails"
+      - string:
+          name: DATE_CLOSED
+          description: "Notify buyers for briefs on a specific closing date (YYYY-MM-DD) - must be earlier than the offset days (28 or 56)"
+      - string:
+          name: BUYER_IDS
+          description: "List of buyer IDs to be emailed e.g. 3,4,6"
+    scm:
+      - git:
+          url: git@github.com:alphagov/digitalmarketplace-scripts.git
+          credentials-id: github_com_and_enterprise
+          branches:
+            - master
+          wipe-workspace: false
+    triggers:
+      - timed: "H 8 * * *"
+    publishers:
+      - trigger-parameterized-builds:
+          - project: notify-slack
+            condition: UNSTABLE_OR_WORSE
+            predefined-parameters: |
+              USERNAME=award-closed-briefs-{{ v.weeks }}-weeks
+              JOB=Notify buyers to award closed briefs after {{ v.weeks }} weeks {{ environment }}
+              ICON=:spaghetti:
+              STAGE={{ environment }}
+              STATUS=FAILED
+              URL=<${BUILD_URL}consoleFull|#${BUILD_NUMBER}>
+              CHANNEL=#dm-release
+    builders:
+      - shell: |
+          [ -d venv ] || virtualenv venv
+
+          . ./venv/bin/activate
+          pip install -r requirements.txt
+
+          if [ -n "$OFFSET_DAYS" ]; then
+            FLAGS="--offset-days=$OFFSET_DAYS"
+          fi
+
+          if [ "$DRY_RUN" = "true" ]; then
+            FLAGS="$FLAGS --dry-run=true"
+          fi
+
+          if [ -n "$DATE_CLOSED" ]; then
+            FLAGS="$FLAGS --date-closed=$DATE_CLOSED"
+          fi
+
+          if [ -n "$BUYER_IDS" ]; then
+            FLAGS="$FLAGS --buyer-ids=$BUYER_IDS"
+          fi
+
+          ./scripts/notify-buyers-to-award-closed-briefs.py {{ environment }} $DM_DATA_API_TOKEN_{{ environment|upper }} $NOTIFY_API_TOKEN $NOTIFY_TEMPLATE_ID $FLAGS
```

Comment:
Can we use clear argument names here? e.g. `./scripts/notify-buyers-to-award-closed-briefs.py --stage={{ environment }} --api-token=$DM_DATA_API_TOKEN_{{ environment|upper }} --govuk-notify-api_key=$NOTIFY_API_TOKEN --govuk_notify_template_id =$NOTIFY_TEMPLATE_ID $FLAGS`

Reply:
Not with the script in its current state - docopt throws an error. We could change the script itself, but the `<stage> <api_key> ...` etc is consistent with what we do in other scripts.

Reply:
Eh, ok. (I really don't like docopt 😛)